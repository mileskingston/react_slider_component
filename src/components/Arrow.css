.slider__arrow {
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 50%;
  width: 50px;
  height: 50px;
  margin-top: -25px;
  background-color: grey;
  border: none;
  cursor: pointer;
  z-index: 999;
  transition: 0.1s;
}
.slider__arrow:hover {
  transform: scale(1.1);
}
.slider__arrow--left {
  left: 0;
}
.slider__arrow--right {
  right: 0
}