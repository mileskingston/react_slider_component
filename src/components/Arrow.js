import React from 'react';
import PropTypes from 'prop-types';

const Arrow = props => (
  <button
    className={`arrw ${props.direction === 'left' ? 'arrw--left' : 'arrw--right'}`}
    onClick={props.onClick}
  >
    {props.direction === 'left' ? 'left' : 'right'}
  </button>
);

Arrow.displayName = 'Arrow';

Arrow.propTypes = {
  direction: PropTypes.string,
  onClick: PropTypes.func.isRequired
};

Arrow.defaultProps = {
  direction: 'left'
};

export default Arrow;
