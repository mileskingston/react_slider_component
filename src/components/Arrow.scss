@import "../_variables.scss";

.slider__arrow {
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 50%;
  width: 50px;
  height: 50px;
  margin-top: -25px;
  background-color: $white;
  border: none;
  cursor: pointer;
  z-index: 999;
  transition: 0.1s;

  &:hover {
    background-color: $grey;
  }

  &--left {
    left: 0;
  }
  &--right {
    right: 0
  }

  & .icon--left,
  & .icon--right {
    width: 30px;
    height: 30px;
  }
}
