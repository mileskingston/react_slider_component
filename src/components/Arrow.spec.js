import React from 'react';
import { mount } from 'enzyme';

import Arrow from './Arrow';

describe('Arrow component', () => {
    it('should render with matching props', () => {
        const mockFunc = jest.fn();
        const wrapper = mount(<Arrow onClick={mockFunc} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should call onClick function on click', () => {
        const mockFunc = jest.fn();
        const wrapper = mount(<Arrow onClick={mockFunc} />);

        wrapper.simulate('click');
        expect(mockFunc).toHaveBeenCalled();
    });

    it('should contain correct class', () => {
        const mockFunc = jest.fn();
        const wrapper = mount(<Arrow onClick={mockFunc} direction="right" />);

        expect(wrapper.find('.slider__arrow--right').length).toBe(1);
    });

    it('should contain correct icon', () => {
        const mockFunc = jest.fn();
        const wrapper = mount(<Arrow onClick={mockFunc} direction="right" />);

        expect(wrapper.find('.icon--right').length).toBe(1);
    });
});