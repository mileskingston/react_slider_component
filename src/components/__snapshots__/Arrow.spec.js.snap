// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow component should render with matching props 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Arrow
    direction="left"
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "direction": "left",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": <button
        class="slider__arrow slider__arrow--left"
      >
        <span
          class="sr--only"
        >
          left
        </span>
        <svg
          class="icon icon--left"
        >
          <use
            xlink:href="#icons_left"
          />
        </svg>
      </button>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="sr--only"
          >
            left
          </span>,
          <Icon
            name="left"
          />,
        ],
        "className": "slider__arrow slider__arrow--left",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <span
            class="sr--only"
          >
            left
          </span>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "left",
            "className": "sr--only",
          },
          "ref": null,
          "rendered": Array [
            "left",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "left",
          },
          "ref": null,
          "rendered": Object {
            "instance": <svg
              class="icon icon--left"
            >
              <use
                xlink:href="#icons_left"
              />
            </svg>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="#icons_left"
              />,
              "className": "icon icon--left",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <use
                  xlink:href="#icons_left"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "xlinkHref": "#icons_left",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "use",
              },
            ],
            "type": "svg",
          },
          "type": [Function],
        },
      ],
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "direction": "left",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": <button
          class="slider__arrow slider__arrow--left"
        >
          <span
            class="sr--only"
          >
            left
          </span>
          <svg
            class="icon icon--left"
          >
            <use
              xlink:href="#icons_left"
            />
          </svg>
        </button>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="sr--only"
            >
              left
            </span>,
            <Icon
              name="left"
            />,
          ],
          "className": "slider__arrow slider__arrow--left",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <span
              class="sr--only"
            >
              left
            </span>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "left",
              "className": "sr--only",
            },
            "ref": null,
            "rendered": Array [
              "left",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "left",
            },
            "ref": null,
            "rendered": Object {
              "instance": <svg
                class="icon icon--left"
              >
                <use
                  xlink:href="#icons_left"
                />
              </svg>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <use
                  xlinkHref="#icons_left"
                />,
                "className": "icon icon--left",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <use
                    xlink:href="#icons_left"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "xlinkHref": "#icons_left",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "use",
                },
              ],
              "type": "svg",
            },
            "type": [Function],
          },
        ],
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
